coordinator/src/main/scala/filodb.coordinator/queryplanner/HighAvailabilityPlanner.scala

```diff
@@ -59,21 +74,28 @@ class HighAvailabilityPlanner(dsRef: DatasetRef,
             (timeRange.startMs + offsetMs) / 1000, queryParams.stepSecs, (timeRange.endMs + offsetMs) / 1000,
             queryParams.spread, processFailure = false)
           logger.debug("PromQlExec params:" + promQlParams)
-          PromQlRemoteExec(remoteHttpEndpoint, remoteHttpTimeoutMs,
-            qContext, InProcessPlanDispatcher, dsRef, promQlParams)
+          val httpEndpoint = remoteHttpEndpoint + queryParams.remoteQueryPath.getOrElse("")
+          rootLogicalPlan match {
+            case lp: LabelValues         => val urlParams = Map("filter" -> lp.filters.map{f => f.column +
+                                           f.filter.operatorString + f.filter.valuesStrings.head}.mkString(","),
+                                           "labels" -> lp.labelNames.mkString(","))
+                                            MetadataRemoteExec(httpEndpoint, remoteHttpTimeoutMs,
+                                            urlParams, qContext, InProcessPlanDispatcher, dsRef, promQlParams)
+            case lp: SeriesKeysByFilters => val urlParams = Map("match[]" -> queryParams.promQl)
+                                            MetadataRemoteExec(httpEndpoint, remoteHttpTimeoutMs,
+                                              urlParams, qContext, InProcessPlanDispatcher, dsRef, promQlParams)
+            case _                       => PromQlRemoteExec(httpEndpoint, remoteHttpTimeoutMs,
+                                            qContext, InProcessPlanDispatcher, dsRef, promQlParams)
```

Comment:
Please align properly

Reply:
improve the readability of this block of code